whatToDad_site/whatToDad_app/views.py

```diff
@@ -33,7 +34,66 @@ def get(self, request):
         )
 
     def post(self, request):
+
         post_form = PostForm(request.POST)
         post_form.save()
 
-        return redirect('home')
\ No newline at end of file
+        return redirect('home')
+
+class ActivityPage(View):
+    def get(self, request):
+
+        activities = Activity.objects.all()
+        activity_form = ActivityForm()
+
+        html_data = {
+            'activities': activities,
+            'form': activity_form,
+        }
+
+        return render(
+            request=request,
+            template_name='activity.html',
+            context= html_data
+        )
+
+    def post(self, request):
+
+        if 'add' in request.POST: 
+            activity_form = ActivityForm(request.POST) 
+            activity_form.is_valid()
+            activity_form.save() 
+
+        return redirect('activity')
+
+        
+class ActivityDetailView(View):
+    def get(self, request, activity_id):
+
+        activity = Activity.objects.get(id=activity_id)
+        # come back to look at the filter method
+        activity_comments = ActivityComments.objects.filter(activity_id=activity_id)
+        activity_comment_form = ActivityCommentForm(activity_object=activity)
+
+        
+        html_data={
+            'activity': activity,
+            'activity_comment_list': activity_comments,
+            'comment_form': activity_comment_form
+        }
+
+        return render(
+            request=request,
+            template_name='activity_detail.html',
+            context= html_data
+        )
+
+    def post(self, request, activity_id):
+        activity = Activity.objects.get(id=activity_id)
+
+        if 'add' in request.POST: 
```

Comment:
If you may not use it at all, consider moving the fetch for the activity to be inside the `if` statement so you don't do unnecessary work when it won't be used